### PR TITLE
fix: checkout action using a github pat

### LINKED ISSUE #357: [Bug]: repository 'https://github.com/***/' not found
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I get an error message when trying to checkout the repository using a custom GitHub PAT.
The PAT used here has the privileges mentionned in the documentation + workflows (rerquired to sync .github).

### Expected Behavior

I would expect the checkout to work.

### Current Behavior

I get this error message:
```
git init
  ::info::set git global configuration
  Updated Git hooks.
  Git LFS initialized.
  ::info::the source repository is located within GitHub.
  github.com
    ✓ Logged in to github.com as dotdc (GITHUB_TOKEN)
    ✓ Git operations for github.com configured to use https protocol.
    ✓ Token: *******************
    
remote: Repository not found.
fatal: repository 'https://github.com/***/' not found
fatal: Needed a single revision
```

### Steps To Reproduce

_No response_

### Possible Solution

Use the PAT to checkout instead of the default token.

### Additional Information/Context

_No response_

### Template sync version Version

v1.0.0-draft

### PR DESCRIPTION
**Pull request by dotdc**
# Description

Close #357

This will fix a checkout issue when using a GitHub PAT.
I also renamed the PAT variable from `SOURCE_REPO_PAT` to `CUSTOM_GITHUB_PAT` in order to avoid match overlap with `SOURCE_REPO_PATH`.

## Remark

For automation please see [closing-issues-using-keywords](
    https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

**Comment by AndreasAugustin** [comment]
@dotdc thanks a lot for the fix.
Did you by any chance forget to push some files related to #357 ?
Actually I only see a documentation change (which makes sense in terms of readability, but I need to think about it in terms of domain namings. Everyone can use a self defined name here.).

**Comment by dotdc** [comment]
Yes documention change only, does anything else require a change ?

Didn't search and replace repo wide, but will check tomorrow morning 👍

**Comment by AndreasAugustin** [comment]
> Yes documention change only, does anything else require a change ?
> 
> Didn't search and replace repo wide, but will check tomorrow morning +1

Ah alright, please let me rephrase your issue with #357 and tell me if I understood right.
You had a typo in the secret or the workflow definition which led to the issue?

Just want to understand if there are code related issues. The part you changed is not related to the code. Just your secret needs the name `CUSTOM_GITHUB_PAT` now, not anymore `SOURCE_REPO_PAT`

**Comment by dotdc** [comment]
> > Yes documention change only, does anything else require a change ?
> > Didn't search and replace repo wide, but will check tomorrow morning +1
> 
> Ah alright, please let me rephrase your issue with #357 and let me know if I got it right You had a typo in the secret or the workflow definition which led to the issue?
> 
> Just want to understand if there are code related issues. The part you changed is not related to the code. Just your secret needs the name `CUSTOM_GITHUB_PAT` now, not anymore `SOURCE_REPO_PAT`

@AndreasAugustin 

No, the issue is that following the current documentation, the workflow will fail with the following error when using a PAT :

```
git init
  ::info::set git global configuration
  Updated Git hooks.
  Git LFS initialized.
  ::info::the source repository is located within GitHub.
  github.com
    ✓ Logged in to github.com as dotdc (GITHUB_TOKEN)
    ✓ Git operations for github.com configured to use https protocol.
    ✓ Token: *******************
    
remote: Repository not found.
fatal: repository 'https://github.com/***/' not found
fatal: Needed a single revision
```

To solve the issue, I had to use the PAT on the checkout action (in this PR change) :

```
        name: Checkout
        uses: actions/checkout@v3
        with:
          token: ${{ secrets.CUSTOM_GITHUB_PAT }}
```

I also proposed to rename the secret name from `SOURCE_REPO_PAT` to `CUSTOM_GITHUB_PAT`, because if you're search for "SOURCE_REPO_PAT" you will also match "SOURCE_REPO_PAT**H**", which is too close to one another, and can lead to confusion.

Just checked, there is no other occurrence of `SOURCE_REPO_PAT` in the repository, so this is only a documentation change.

**Comment by AndreasAugustin** [comment]
Hi @dotdc got it.
Am super sorry. I had been super busy and was reading the PR in a rush. Somehow I did not read the full change..

Makes absolutely sense. Thanks a lot!

**Comment by AndreasAugustin** [comment]
@all-contributors please add @dotdc  for bug

**Review by AndreasAugustin — APPROVED** [review]
